Feature #3307 quality_marker_thresh (take 2)

JohnHalleyGotway opened the pull request:
## Expected Differences ##

- [x] Do these changes introduce new tools, command line arguments, or configuration file options? **[Yes]**</br>
If **yes**, please describe:</br>
Enables PB2NC config option `quality_mark_thresh` to be defined as a threshold rather than just an integer value upper limit.

- [x] Do these changes modify the structure of existing or add new output data types (e.g. statistic line types or NetCDF variables)? **[No]**</br>
If **yes**, please describe:</br>

## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Manually tested with `quality_mark_thresh = <=2||==9;` with the data listed in #3307 to confirm that quality marks of 8 are excluded. The counts of ADPSFC observations by quality marker changed as follows:
  - 2: FROM 80130 TO 80130 (*UNCHANGED*)
  - 3: FROM 3096 TO 0
  - 8: FROM 3814 TO 0
  - 9: FROM 317098 TO 317098 (*UNCHANGED*)
  - 14: FROM 10738 TO 0
  - 15: FROM 2349 TO 0

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Confirm all the unit tests run and no diffs are flagged, even after the majority of the quality mark threshold settings have changed from an integer (e.g. `2`) to a threshold (e.g. `<=2`).

Please find the code compiled and available for additional testing in `seneca:/d1/projects/MET/MET_pull_requests/met-13.0.0/beta1/MET-feature_3307_quality_marker_thresh_take_2`.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the MET test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Will this PR result in changes to existing METplus Use Cases? **[No]**</br>
If **yes**, create a new **Update Truth** [METplus issue](https://github.com/dtcenter/METplus/issues/new/choose) to describe them.

- [ ] Do these changes introduce new SonarQube findings? **[Yes or No]**</br>
If **yes**, please describe:
None expected

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the version that will include these changes
Select: **METplus-X.Y Support** project for bugfix releases or **MET-X.Y Development** project for the next coordinated release
- [x] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

JohnHalleyGotway committed:
Port over changes for #3307 to a new feature branch since GitHub is not recognizing all the changes for my PR.

michelleharrold reviewed (dismissed):
I tested the changes on Seneca using a NAM prepbufr file, running through several iterations of changing the quality_mark_thresh. The output looked good and as expected.

I found the example in the PR helpful (quality_mark_thresh = <=2||==9;) -- I sometimes struggle with syntax. It might be nice to include an example of this in the documentation, e.g.,: This can be specified as a threshold string (<=2||==9) or as an integer defining the maximum allowable quality mark value. Earlier versions of MET only supported the integer setting. Take or leave!

JohnHalleyGotway committed:
Enhance quality_mark_thresh documentation

Clarified the explanation of quality_mark_thresh to include threshold string examples.

JohnHalleyGotway commented:
Thanks for the tip. I updated the docs with that example and will merge now.
